File: model/monetization/SubscriptionPaymentModel.py
class SubscriptionPaymentModel:
    def __init__(self):
        self.paymentIssueDate = None  # LocalDate  # This might be an enum
        self.paymentValueDate = None  # LocalDate  # This might be an enum
        self.amount = None  # String
        self.currency = None  # String
        self.invoiceNo = None  # String

File: model/monetization/SubscriptionPaymentModel.java
package com.digitinary.dgate.model.monetization;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDate;

/**
 * Subscription payment details model.
 *
 * @author Salah Abu Msameh
 * @since 16/08/2024
 */
@NoArgsConstructor
@AllArgsConstructor
@Data
public class SubscriptionPaymentModel {

    private LocalDate paymentIssueDate;
    private LocalDate paymentValueDate;
    private String amount;
    private String currency;
    private String invoiceNo;
}
